cmd/k8s-operator/svc-for-pg.go

```
@@ -0,0 +1,899 @@
+	pod := &corev1.Pod{}
+	err := a.Get(ctx, client.ObjectKey{Namespace: a.tsNamespace, Name: replicaName}, pod)
+	if apierrors.IsNotFound(err) {
+		logger.Debugf("Pod %q not found", replicaName)
+		return false, nil
+	}
+	if err != nil {
+		return false, fmt.Errorf("failed to get Pod: %w", err)
+	}
+	secret := &corev1.Secret{}
+	err = a.Get(ctx, client.ObjectKey{Namespace: a.tsNamespace, Name: replicaName}, secret)
+	if apierrors.IsNotFound(err) {
+		logger.Debugf("Secret %q not found", replicaName)
+		return false, nil
+	}
+	if err != nil {
+		return false, fmt.Errorf("failed to get Secret: %w", err)
+	}
+	if len(secret.Data) == 0 || secret.Data[ingressservices.IngressConfigKey] == nil {
+		return false, nil
+	}
+	gotCfgB := secret.Data[ingressservices.IngressConfigKey]
+	var gotCfgs ingressservices.Status
+	if err := json.Unmarshal(gotCfgB, &gotCfgs); err != nil {
+		return false, fmt.Errorf("error unmarshalling ingress config: %w", err)
+	}
+	statusUpToDate, err := isCurrentStatus(gotCfgs, pod, logger)
+	if err != nil {
+		return false, fmt.Errorf("error checking ingress config status: %w", err)
+	}
+	if !statusUpToDate {
+		logger.Debugf("Pod %q is not ready to advertise VIPService", pod.Name)
+		return false, nil
+	}
+	if !reflect.DeepEqual(gotCfgs.Configs.GetConfig(serviceName), wantsCfg) {
+		wantj, err := json.Marshal(wantsCfg)
+		if err != nil {
+			return false, fmt.Errorf("error marshalling ingress config: %w", err)
+		}
+
+		logger.Debugf("Pod %q has config %q, but wants %q", pod.Name, string(gotCfgB), wantj)
+		return false, nil
+	}
+	return true, nil
+}
+
+func isCurrentStatus(gotCfgs ingressservices.Status, pod *corev1.Pod, logger *zap.SugaredLogger) (bool, error) {
+	ips := pod.Status.PodIPs
+	if len(ips) == 0 {
+		logger.Debugf("Pod %q does not yet have IPs, unable to determine if status is up to date", pod.Name)
+		return false, nil
+	}
+
+	if len(ips) > 2 {
+		return false, fmt.Errorf("POD_IPs can contain at most 2 IPs, got %d (%v)", len(ips), ips)
```

I think in this case we don't want to return an error that would cause a reconcile? I don't think the number of Pod IPs would have changed on reconcile